libs/domains/application/src/lib/slices/applications.slice.ts

```diff
@@ -74,11 +80,61 @@ const containerMetricsApi = new ContainerMetricsApi()
 const containerDeploymentsApi = new ContainerDeploymentHistoryApi()
 const containerConfigurationApi = new ContainerConfigurationApi()
 
+const databasesApi = new DatabasesApi()
+
 const jobsApi = new JobsApi()
 const jobMainCallsApi = new JobMainCallsApi()
 const jobDeploymentsApi = new JobDeploymentHistoryApi()
 const jobConfigurationApi = new JobConfigurationApi()
 
+type UseCloneServiceProps =
+  | {
+      serviceId: string
+      serviceType: ServiceTypeEnum.APPLICATION
+      cloneRequest: CloneApplicationRequest
+    }
+  | {
+      serviceId: string
+      serviceType: ServiceTypeEnum.CONTAINER
+      cloneRequest: CloneContainerRequest
+    }
+  | {
+      serviceId: string
+      serviceType: ServiceTypeEnum.DATABASE
+      cloneRequest: CloneDatabaseRequest
+    }
+  | {
+      serviceId: string
+      serviceType: ServiceTypeEnum.JOB | ServiceTypeEnum.LIFECYCLE_JOB | ServiceTypeEnum.CRON_JOB
+      cloneRequest: CloneJobRequest
+    }
+
+export const useCloneService = () => {
```

Comment:
I think it's better to have an file `services.queries.ts` on a library `Service` with this mutation because you have database actions on it

Reply:
https://github.com/Qovery/console/pull/700/commits/41c64cd69492aa24e54863c7858a22f0ff798d7e